### PR TITLE
Adding documentation on prefetching  in Sync Receive API

### PR DESCRIPTION
**Pull request by anuchandy**
The documentation explains how synchronous `receiveMessages` API deals with message prefetching.

### CODE REVIEW

File: sdk/servicebus/azure-messaging-servicebus/docs/SyncReceiveAndPrefetch.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
+### The Sync receiveMessages API and implicit Prefetch:
```
[the remark this answers is not in the document]
Comment (conniey) [reply sdk/servicebus/azure-messaging-servicebus/docs/SyncReceiveAndPrefetch.md:1]: I think the headers should be at level 1 (ie. `# Sync ``receiveMessages`` API and implicit prefetch`) and the subsequent ones at level 2 (ie. `##`)

I think it makes it more obvious what each section focuses on, too. :)

File: sdk/servicebus/azure-messaging-servicebus/docs/SyncReceiveAndPrefetch.md [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+Given the prefetch-queue is shared, the messages will get asynchronously buffered even if:
+1. the application ignores the Iterable returned from `receiveMessages` calls.
+2. the application stops iteration in the middle.
+3. the `maxTimeout` elapses before receiving `maxMessages`.
+
+> On #3, the current iterator will be completed if the `maxTimeout` elapses, but the buffering continues in the background.
+
+Any messages from a previous `receiveMessages` call that its Iterable couldn't read from the buffer (prefetch-queue) will be read and delivered by the Iterable from the next `receiveMessages` call on the same Client. 
+
+Let's take the previous example, say within `maxTimeout` of 5 sec, the service returned only 7 messages out of 10; in this case, the `messagesIterable` will complete after returning those 7 messages and exit the for loop. The remaining 3 messages can get buffered in the background. If the application again calls `receiveMessages(5, ..)`, then SDK initiates a request for 5 messages. The new Iterable read and deliver 5 messages from the buffer (starting with the first three messages buffered by initial `revceiveMessages` call).
```
Comment (conniey) [inline sdk/servicebus/azure-messaging-servicebus/docs/SyncReceiveAndPrefetch.md:60]:
```suggestion
Let's take the previous example, say within `maxTimeout` of 5 sec, the service returned only 7 messages out of 10; the `messagesIterable` will complete after returning those 7 messages and exits the for-loop. The remaining 3 messages can get buffered in the background. If the application then calls `receiveMessages(5, ..)`, then SDK initiates a request for 5 messages. The new Iterable read and deliver 5 messages from the buffer (starting with the first three messages buffered by the initial `receiveMessages` call).
```

**Review by conniey — APPROVED** [review]

File: sdk/servicebus/azure-messaging-servicebus/docs/SyncReceiveAndPrefetch.md [hunk 3]
```diff
@@ -0,0 +1,85 @@
+### The Sync receiveMessages API and implicit Prefetch:
+
+The documentation [here][prefetchgeneral] describes the Service Bus SDK prefetch feature in general. It also states that setting the prefetch-count property in the builder to zero disables prefetch.
```
Comment (conniey) [inline sdk/servicebus/azure-messaging-servicebus/docs/SyncReceiveAndPrefetch.md:3]: We could add a link to this doc in azure-messaging-servicebus/README.md under the troubleshooting guide.

Comment (anuchandy) [reply sdk/servicebus/azure-messaging-servicebus/docs/SyncReceiveAndPrefetch.md:3]: Good idea! I'll do that after merging this pr to ensure the referenced file is in place.

Comment (anuchandy) [reply sdk/servicebus/azure-messaging-servicebus/docs/SyncReceiveAndPrefetch.md:3]: addressed in the pr: https://github.com/Azure/azure-sdk-for-java/pull/25885

**Pull request merged**